Add invalid recipients behaviour and improve direct socket description to readme (#168)
samply/beam: README.md

[![Build with rust and docker](https://github.com/samply/beam/actions/workflows/rust.yml/badge.svg)](https://github.com/samply/beam/actions/workflows/rust.yml)

> Samply.Beam is a distributed task broker designed for efficient communication across strict network environments. It provides most commonly used communication patterns across strict network boundaries, end-to-end encryption and signatures, as well as certificate management and validation on top of an easy to use REST API. For especially low-level or demanding applications, Samply.Beam can provide encrypted Direct-Socket connections.

Reviewer: ```suggestion
Samply.Beam is a distributed task broker designed for efficient communication across strict network environments. It provides most commonly used communication patterns across strict network boundaries, end-to-end encryption and signatures, as well as certificate management and validation on top of an easy to use REST API. In addition to task/response semantics, Samply.Beam supports high-performance applications with encrypted low-level direct socket connections.
```

This *long polling* opens the connection and sleeps until a reply is received. For more information, see the API documentation.

### Using direct socket connections
> Only available on builds of beam with the `sockets` feature 

> The establishing of Direct-Sockets via beam requires a negotiation phase prior to using the sockets. One application sends a socket request to the other application via their respective Beam.Proxies. The receiving application can polls or wait for incoming socket requests. Once they are accepted, the connection is upgraded to a TCP-based, encrypted socket connection.

Reviewer: ```suggestion
Establishing direct socket connections via Beam requires a negotiation phase prior to using the sockets. One application sends a socket request to the other application via their respective Beam.Proxy. The receiving application, upon receipt of the request, upgrades the connection to an encrypted TCP socket connection.
```

> While possible to show the usage of Samply.Beam-Sockets similarly to the task-based usage above, that is making use of command line tools, such as curl, netcat, or socat, the intended usage is programmatically by the applications' code. Thus, we show the usage in the following short python application exemplifying both applications, the initiating and the receiving one. Both sides of the communication run concurrently.

Reviewer: ```suggestion
While Beam sockets can be initiated using command line tools, such as curl, netcat, or socat, they are intended to be used by the applications' code. Thus, we show the usage in the following short python application exemplifying both applications, the initiating and the receiving one. Both sides of the communication run concurrently.
```
